examples/minimal_client_service/src/minimal_client_async.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::{Error, Result};
+use std::env;
+
+#[tokio::main]
```

Comment:
> This is the thing that I fear is unidiomatic for Rust

Do you have a source for this?

It seems to me that this is a completely valid case, given that tokio itself has a specific API for this and even the example for it is quite similar:

https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html

In any case, this is an example of how to integrate the `rclrs` event loop with tokio, users can adapt it to their liking. Other libraries (like r2r), follow a similar pattern (https://github.com/sequenceplanner/r2r/blob/master/examples/client.rs#L42-L52)

Reply:
> Do you have a source for this?

I don't have one that addresses this head-on, but for instance the official async Rust book, particularly [the beginning of this chapter](https://rust-lang.github.io/async-book/02_execution/04_executor.html), sounds like it's the responsibility of the async executor (and only the async executor) to run them to completion. (Side note: I'm not saying we should build our own async executor)

I also asked this on the Rust discord whether it's unidiomatic to design a Future such that it can not be driven to completion by an async executor alone, but needs a separate function to run in the background to make progress. Just got one answer, which was "I'd probably be surprised by it, but it's not in principle very different from 'function deadlocks unless certain conditions, please satisfy those conditions' ".  I think the difference here is that a deadlock condition probably cannot be solved with a different design, but we *could* design the future in a way that the separate spin is not needed.

The purpose of `spawn_blocking` is to perform a heavy piece of blocking (=non-async) work inside a `Future`, I don't think it's related to the current question (or at least it needs to be explained to me how it is).

All in all, I still think it's not terrible, but maybe a bit weird. We can merge this and continue in a GitHub discussion about async stuff.